### DISCUSSION TITLE
Compactor is not deleting raw data after the specified retention

### DISCUSSION THREAD

**Discussion by anoop2503** [thanos-io/thanos#5449] [Questions & Answers]
Hi,

I am running thanos compactor with below configuration for more than a year. Based on the below configuration, I was assuming that the raw data which are older than 180 days will be deleted automatically and only downsampled data will be available to view, if you check the data older than 180 days. But, when I query, I can still see raw data for more than 250 days. I am using the thanos version: v0.23.1 now,

Compactor configuration:

```
- args:
        - compact
        - --log.level=debug
        - --data-dir=/compact
        - --objstore.config-file=/etc/thanos-config/thanos-s3-config.yml
        - --retention.resolution-raw=180d
        - --retention.resolution-5m=365d
        - --retention.resolution-1h=0d
        - --wait
        - --http-address=0.0.0.0:10902
```

When I checked the thanos-compactor logs, it is showing the logs for deleting the blocks.

```
level=info ts=2022-05-27T11:16:44.428028Z caller=blocks_cleaner.go:53 msg="deleted block marked for deletion" block=01EQAG1JF04WZKP2EX7DQC9473
level=debug ts=2022-05-27T11:16:44.599439734Z caller=block.go:233 msg="deleted file" file=01EQA2A3Z0Q8SPA8A6RFY0PZBX/meta.json bucket="tracing: abcd-efg-hij-klm"
level=debug ts=2022-05-27T11:16:44.793293067Z caller=block.go:276 msg="deleted file" file=01EQA2A3Z0Q8SPA8A6RFY0PZBX/index bucket="tracing: abcd-efg-hij-klm"
level=debug ts=2022-05-27T11:16:45.004983006Z caller=block.go:276 msg="deleted file" file=01EQA2A3Z0Q8SPA8A6RFY0PZBX/chunks/000001 bucket="tracing: abcd-efg-hij-klm"
level=debug ts=2022-05-27T11:16:45.135644526Z caller=block.go:276 msg="deleted file" file=01EQA2A3Z0Q8SPA8A6RFY0PZBX/chunks/000002 bucket="tracing: abcd-efg-hij-klm"
level=debug ts=2022-05-27T11:16:45.337284365Z caller=block.go:256 msg="deleted file" file=01EQA2A3Z0Q8SPA8A6RFY0PZBX/deletion-mark.json bucket="tracing: abcd-efg-hij-klm"
```

So, I am confused why it is still showing the raw data? Can someone please confirm, if there is something that I am missing in the configuration? Do I need to check some other components to find out the root cause?

Thanks,

**Comment by vaillani** [comment]
Hello here !

i'm actually facing the same issue with a compactor retention :

```yaml
- --retention.resolution-raw=365d
- --retention.resolution-5m=365d
- --retention.resolution-1h=365d
```

The compactor logs shows that block marked for deletion has finished but when I inspect the bucket, there is still block dated 2020-`04-05-2020 00:00:00`. Does anybody have a idea to fix this behaviour ?

Thanks in advance.
